src/main/java/modify_event_use_case/ModifyEventInteractor.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Interactor for the Modify Event use case. Calls the dsGateway and Output Boundary to modify the event.
+ * Author: Daniel Livshits
+ * Created: Nov 19, 2022
+ * Modified by: Daniel Livshits
+ * Last Modified: Nov 27, 2022
+ */
+package modify_event_use_case;
+
+import java.time.LocalTime;
+
+public class ModifyEventInteractor implements ModifyEventInputBoundary{
+    private final ModifyEventOutputBoundary outputBoundary;
+    private final ModifyEventDsGateway dsGateway;
+
+    private final String[] DAYSOFWEEK = {"Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"};
+
+    public ModifyEventInteractor(ModifyEventOutputBoundary outputBoundary, ModifyEventDsGateway dsGateway){
+        this.outputBoundary = outputBoundary;
+        this.dsGateway = dsGateway;
+    }
+
+    /**
+     * Modifies the desired event by making calls to the outputBoundary and dsGateway, if there are no conflicts.
+     * If there is a naming conflict or time conflict, creates an instance of Output Data representing the corresponding
+     * failure state.
+     * @param inputData - the data required to modify the desired event.
+     * @return an instance of output data, containing the necessary information about the success or failure of
+     * the modification.
+     */
+    @Override
+    public ModifyEventOutputData modify(ModifyEventInputData inputData){
+        if(!(inputData.getNewStartTime().matches("[01][0-9]:[0-5][0-9]|2[0-3]:[0-5][0-9]"))) {
+            ModifyEventOutputData outputData = new ModifyEventOutputData(inputData.getTitle(), inputData.getDayIndex(),
+                    inputData.getNewTitle(), null, null);
+            String failMessage = "The start and end times are not valid times!";
+            return outputBoundary.prepareFailView(outputData, failMessage);
+        }
+        LocalTime newStartTime = LocalTime.parse(inputData.getNewStartTime());
+        LocalTime newEndTime = LocalTime.parse(inputData.getNewEndTime());
+        ModifyEventOutputData outputData = new ModifyEventOutputData(inputData.getTitle(), inputData.getDayIndex(),
+                inputData.getNewTitle(), newStartTime, newEndTime);
+        if(newStartTime.isAfter(newEndTime) || newStartTime.equals(newEndTime)){
+            String failMessage = "The new start time is not before the new end time!";
+            return outputBoundary.prepareFailView(outputData, failMessage);
+        }
+        if(!(inputData.getTitle().equals(inputData.getNewTitle())) &&
```

Comment:
Why exactly is title being compared against newTitle? I feel it would suffice to just pass your new title to the gateway and get it to check whether there's an event with that title for the given day.

Reply:
This is for the case where the user only wants to change the times, not the title, for the event. If the old title and new title are the same, there would be an event with the newTitle in the day already: the exact event we are trying to modify. So having the gateway check for an event with the same name would return true, which causes the modification to fail, which is not the desired result.

Reply:
Well in that case, the gateway should just check if the new times overlap with the times for any other Event for that day, which does not have the same name as the Event being modified.

Reply:
### resolved in conversation outside of GitHub.